cumulusci/core/flowrunner.py

```diff
@@ -0,0 +1,428 @@
+        task_class = import_class(task_config["class_path"])
+
+        # TODO: Resolve ^^task_name.return_value style option syntax
+        exc = None
+        try:
+            task = task_class(
+                self.project_config,
+                TaskConfig(task_config),  # BaseTask wants a full on TaskConfig
+                org_config=self.org_config,
+                name=step.task_name,
+                stepnum=step.step_num,
+                flow=self,  # not actually passing the flow, but that doesn't matter, this is more a run_in_flow.
+                # TODO: fix that!
+            )
+            task()
+        except Exception as exc:
+            # David, I don't know if this makes any sense.
+            task.logger.exception("Exception in task {}".format(step.task_name))
+        finally:
+            return StepResult(
+                step.step_num, step.task_name, task.result, task.return_values, exc
+            )
+
+
+class FlowCoordinator(object):
+    def __init__(self, project_config, flow_config, name=None, options=None, skip=None):
+        self.project_config = project_config
+        self.flow_config = flow_config
+        self.name = name
+        self.org_config = None
+
+        # TODO: Support CLI/Runtime Options
+        if not options:
+            options = {}
+        self.runtime_options = options
+
+        if not skip:
+            skip = []
+        self.skip = skip
+
+        self.results = []
+
+        self.logger = self._init_logger()
+        self.steps = self._init_steps()  # type: List[StepSpec]
+
+    def _rule(self, fill="=", length=60, new_line=False):
+        self.logger.info("{:{fill}<{length}}".format("", fill=fill, length=length))
+        if new_line:
+            self.logger.info("")
+
+    def run(self, org_config):
+        self.org_config = org_config
+        line = "Initializing flow: {}".format(self.__class__.__name__)
+        if self.name:
+            line = "{} ({})".format(line, self.name)
+        self._rule()
+        self.logger.info(line)
+        self._rule(new_line=True)
+        self._init_org()
+
+        self._pre_flow()
+
+        self._rule(fill="-")
+        for step in self.steps:
+            self.logger.info(step.for_display)
+        self._rule(fill="-", new_line=True)
+
+        self.logger.info("Starting execution")
+        self._rule(new_line=True)
+
+        try:
+            for step in self.steps:
+                if step.skip:
+                    self._rule(fill="*")
+                    self.logger.info("Skipping task: {}".format(step.task_name))
+                    self._rule(fill="*", new_line=True)
+                    continue
+
+                self._rule(fill="-")
+                self.logger.into("Running task: {}".format(step.task_name))
+                self._rule(fill="-", new_line=True)
+
+                self._pre_task(step)
+                runner = TaskRunner(self.project_config, org_config=org_config)
```

Comment:
Once I work on embedding steps from other projects I think the project_config will need to be referenced from the StepSpec.